lib/streamlit/elements/markdown.py

```diff
@@ -271,6 +277,85 @@ def divider(self) -> DeltaGenerator:
         divider_proto.element_type = MarkdownProto.Type.DIVIDER
         return self.dg._enqueue("markdown", divider_proto)
 
+    @gather_metrics("badge")
+    def badge(
+        self,
+        label: str,
+        *,  # keyword-only arguments:
+        icon: str | None = None,
+        color: Literal[
+            "blue",
+            "green",
+            "orange",
+            "red",
+            "violet",
+            "gray",
+            "grey",
+            "rainbow",
+            "primary",
+        ] = "blue",
+    ) -> DeltaGenerator:
+        """Display a colored badge with an icon and label.
+
+        You can also insert badges directly in Markdown, e.g. via
+        `st.markdown(":blue-badge[Home]")`. This works in all places where Streamlit
+        supports Markdown, e.g. widget labels or `st.table` cells. See `st.markdown`
+        for more information.
+
+        Parameters
+        ----------
+        label : str
+            The text to display in the badge.
```

Comment:
Should we mention here that the label does support limited markdown as well?

Reply:
Added. Had to do a bit of trial and error which things are supported since there's no good information online for remark-directive.